lib/codelog/command/step/version.rb

```diff
@@ -11,24 +11,25 @@ class Version
         CONFIG_FILE_PATH = 'changelogs/codelog.yml'.freeze
         RELEASES_PATH = 'changelogs/releases'.freeze
 
-        def initialize(version, release_date)
+        def initialize(version, release_date, outputter)
           abort(Codelog::Message::Error.missing_config_file) unless config_file_exists?
           @version = version
           @release_date = Date.strptime(release_date, Codelog::Config.date_input_format).to_s
+          @outputter = outputter
         rescue ArgumentError
           abort(Codelog::Message::Error.invalid_date_format)
         end
 
-        def self.run(version, release_date)
-          Codelog::Command::Step::Version.new(version, release_date).run
+        def self.run(version, release_date, outputter)
+          Codelog::Command::Step::Version.new(version, release_date, outputter).run
         end
 
         def run
           abort(Codelog::Message::Error.missing_version_number) if @version.nil?
           abort(Codelog::Message::Error.already_existing_version(@version)) if version_exists?
           abort(Codelog::Message::Error.no_detected_changes(@version)) unless unreleased_changes?
 
-          generate_changelog_content_from(changes_hash)
+          @outputter.print generate_changelog_content_from(changes_hash)
```

Comment:
@emilianoLeite Now, Dependency injection is possible, thanks to the changes made on this commit. A small tradeoff is that in order to this to work, the passed outputter must be ready to do what it has to do only by calling the `#print` method passing the desired content.